File: 1603. Design Parking System.java
/**
 * https://leetcode.com/problems/design-parking-system/
 */

// Method 1
class ParkingSystem {
    private int[] count;

    public ParkingSystem(int big, int medium, int small) {
        count = new int[]{big, medium, small};
    }

    public boolean addCar(int carType) {
        return count[carType - 1]-- > 0;
    }
}

// Method 2
class ParkingSystem {
    private int big, medium, small;

    public ParkingSystem(int big, int medium, int small) {
        this.big = big;
        this.medium = medium;
        this.small = small;
    }

    public boolean addCar(int carType) {
        boolean res = false;
        switch (carType) {
            case 1:
                if (this.big-- > 0)
                    res = true;
                break;
            case 2:
                if (this.medium-- > 0)
                    res = true;
                break;
            case 3:
                if (this.small-- > 0)
                    res = true;
                break;
        }
        return res;
    }
}

File: Kotlin/1603. Design Parking System.kt
/**
 * https://leetcode.com/problems/design-parking-system/
 */

class ParkingSystem(big: Int, medium: Int, small: Int) {
    private var big = 0
    private var medium = 0
    private var small = 0

    init {
        this.big = big
        this.small = small
        this.medium = medium
    }

    fun addCar(carType: Int): Boolean {
        when (carType) {
            1 -> return big-- > 0
            2 -> return medium-- > 0
            3 -> return small-- > 0
        }
        return false
    }

}